operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -841,6 +867,10 @@ func (r *StackReconciler) markStackSucceeded(ctx context.Context, instance *pulu
 		LastSuccessfulCommit: current.Commit,
 		Permalink:            shared.Permalink(update.Status.Permalink),
 		LastResyncTime:       metav1.Now(),
+		Attempts:             int64(1),
+	}
+	if last != nil && last.LastAttemptedCommit == current.Commit {
+		instance.Status.LastUpdate.Attempts += last.Attempts
```

Comment:
I see, in the periodic resync, we'd see attempts increment at each run.  I think that is mildly confusing and is overkill.  I would keep it at 1.